### PR TITLE
Gem puzzle

### PR DESCRIPTION
**Pull request by andrei-roh**
Ссылка на задание.
https://github.com/rolling-scopes-school/tasks/blob/master/tasks/gem-pazzle/codejam-the-gem-puzzle.md
Скриншот результата выполнения задания:
Снимок экрана 2020-11-28 224042
Ссылка на задеплоенную версию приложения:
https://andrei-roh-gem-puzzle.netlify.app/
Дата сдачи: 15.11.2020 / дата дедлайна: 16.11.2020.
Самопроверка с предварительной оценкой: 100 баллов

### CODE REVIEW

File: js/boards/bigBoard.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var bigBoard = document.getElementById('6x6');
+bigBoard.addEventListener ('click', () => {
+	startBoard = {
+		setRows: 6,
```
Comment (ollybondareva) [inline js/boards/bigBoard.js:4]: not the best name. why not simply use rows / columns etc?
setRows may think me about function..

**Review by ollybondareva — COMMENTED** [review]
Please, write your code in more accurate way, use linters and formatters, try to make your code more readable, avoid strange names. 
Split big functions into smaller once.
Try next time to use OOP and classes.

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      justify-content: space-around;
+  width: 500px;
+  -ms-flex-item-align: center;
+      align-self: center;
+}
+#reset {
+	border: solid 1px white;
+	background-color: #4b475c;
+	margin-top: 5px;
+	border: none
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:58]: OK, I understood

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+.victory {
+  font-size: 2em;
+}
+.cross {
+  align-self: flex-end;
+  display: inherit;
+  border: 2px solid white;
+  -webkit-box-sizing: border-box;
+          box-sizing: border-box;
+  display: block;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:209]: OK, I understood

File: css/style.css [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  width: 40px;
+  height: 40px;
+  border-radius: 50%;
+  line-height: 40px;
+  text-align: center;
+  -ms-flex-item-align: end;
+      -ms-grid-row-align: end;
+      align-self: flex-end;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:217]: OK, I understood

File: css/style.css [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  margin: 0.5em 0.5em 0.5em 0;
+}
+ul {
+  margin-left: 0px;
+}
+#centralBox {
+  display: -webkit-box;
+  display: -ms-flexbox;
+  display: flex;
+  -webkit-box-orient: vertical;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:29]: OK, I understood

File: css/menu_left.css [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+.nav-toggle_left:hover {
+    color: black;
+}
+
+/*Hidden checkbox*/
+[id='nav-toggle_left'] {
+    position: absolute;
+    display: none;
+}
+/*if (panel open){.nav change}*/
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/menu_left.css:48]: OK, I understood

File: css/menu_left.css [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    box-shadow:4px 0px 20px 0px rgba(0,0,0, 0.5);
+    -moz-box-shadow:4px 0px 20px 0px rgba(0,0,0, 0.5);
+    -webkit-box-shadow:4px 0px 20px 0px rgba(0,0,0, 0.5);
+    overflow-y: auto;
+}
+/*MULTIPLICATION X*/
+[id='nav-toggle_left']:checked ~ .nav_left > .nav-toggle:after {
+    content: '\2715';
+}
+/*плавное появление заголовка (логотипа) при раскрытии панели */
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/menu_left.css:60]: OK, I understood

File: css/style.css [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    outline: none;
+		cursor: pointer;
+}
+p {
+  margin: 0.5em 0.5em 0.5em 0;
+}
+ul {
+  margin-left: 0px;
+}
+#centralBox {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:25]: OK, I understood

File: css/style.css [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+}
+.time {
+	font-size: 16px;
+}
+.logo_header {
+  padding-left: 38px;
+}
+
+/* POPUP style */
+.b-popup{
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:120]: OK, I understood

File: css/style.css [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+    overflow: hidden;
+    position: absolute;
+    top: 0px;
+    padding-top: 0px;
+    z-index: 3000;
+  }
+}
+@media only screen and (min-width: 1200px){
+  .b-popup {
+    /* width: 100%; */
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:144]: OK, I understood

File: css/menu_right.css [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    color: black;
+}
+
+/*Hidden checkbox*/
+[id='nav-toggle_right'] {
+    position: absolute;
+    display: none;
+}
+
+[id='nav-toggle_right']:checked ~ .nav_right > .nav-toggle_right {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/menu_right.css:49]: OK, I understood

File: css/style.css [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+button:active,
+button:focus {
+    outline: none;
+		cursor: pointer;
+}
+p {
+  margin: 0.5em 0.5em 0.5em 0;
+}
+ul {
+  margin-left: 0px;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/style.css:23]: OK, I understood

File: css/menu_right.css [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+}
+/*make button icon (TRIGRAM FOR HEAVEN)*/
+.nav-toggle_right:after {
+    content: '\2630';
+    text-decoration: none;
+}
+
+/*text color on hover*/
+.nav-toggle_right:hover {
+    color: black;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply css/menu_right.css:40]: OK, I understood

File: js/boards/bigBoard.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+var bigBoard = document.getElementById('6x6');
+bigBoard.addEventListener ('click', () => {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/boards/bigBoard.js:2]: OK, I understood

File: js/boards/biggestBoard.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+var biggestBoard = document.getElementById('7x7');
+biggestBoard.addEventListener ('click', () => {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/boards/biggestBoard.js:2]: OK, I understood

File: js/cutter.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
+//add random image from images folder
+const imagesFolder = 'assets/images/';
+var quantity = 150;
+let images = Array.apply(null, {length: quantity}).map(Number.call, Number);
+images.shift();
+var imageSrc = 'url' + '(' + imagesFolder + images[randomInteger(1, 149)] + '.jpg' + ')';
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/cutter.js:6]: OK, I understood

File: js/game.js [hunk 17]
```diff
@@ -0,0 +1,299 @@
+//create body
+var body = document.querySelector("body");
+//create left menu
+var inputLeft = document.createElement("input");
+var nav_left = document.createElement("nav");
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:5]: OK, I understood

File: js/game.js [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+//create popup
+var b_popup = document.createElement("div");
+var b_popup_content = document.createElement("div");
+var b_popup_box = document.createElement("div");
+var a_popup = document.createElement("a");
+var victory_popup = document.createElement("div");
+
+
+//filling left menu
+inputLeft.id = "nav-toggle_left";
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:43]: OK, I understood

File: js/game.js [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+inputRight.hidden = "";
+nav_right.classList.add("nav_right");
+nav_right.innerHTML = "<label for='nav-toggle_right' class='nav-toggle_right' onclick></label>";
+h2_right.classList.add("logo");
+scoreButton.innerHTML = '<button class="menu_button" id="reset" type="button" name="button">Score</button>';
+top_score.innerHTML = '<div class="top_score" id="top_score"></div>'
+//filling central box
+centralBox.id = "centralBox";
+infoBox.id = "infoBox";
+box.id = 'box';
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:72]: OK, I understood

File: js/game.js [hunk 20]
```diff
@@ -0,0 +1,299 @@
...
+form.name = 'myForm';
+span.id = 'time';
+inputWatch.id = 'watch';
+inputWatch.name = 'stopwatch';
+inputWatch.size = '10';
+inputWatch.value = '00:00:00.00'
+inputWatch.disabled = '';
+moves.id = 'moves';
+//filling popup
+b_popup.classList.add("b-popup");
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:82]: OK, I understood

File: js/game.js [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+b_popup_box.appendChild(victory_popup);
+
+
+var arr = [], emptyRow, emptyColumn;
+var playerMoves = 0;
+var score = [];
+var victoryMessage = document.getElementById('victory');
+var top_score = document.getElementById('top_score')
+var watchTime = document.getElementById('watch')
+moves.innerHTML =`Moves: 0`;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:137]: OK, I understood

File: js/game.js [hunk 22]
```diff
@@ -0,0 +1,299 @@
...
+infoBox.appendChild(moves);
+//placement popup box
+document.body.appendChild(b_popup);
+b_popup.appendChild(b_popup_content);
+b_popup_content.appendChild(b_popup_box);
+b_popup_box.appendChild(a_popup);
+b_popup_box.appendChild(victory_popup);
+
+
+var arr = [], emptyRow, emptyColumn;
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:131]: OK, I understood

File: js/game.js [hunk 23]
```diff
@@ -0,0 +1,299 @@
...
+time.innerHTML =`Time: `;
+
+//keysound
+function soundHandler(soundSource) {
+	const keySound = document.createElement('audio');
+	keySound.src = soundSource;
+	keySound.play();
+ }
+
+let startBoard = {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:147]: OK, I understood

File: js/game.js [hunk 24]
```diff
@@ -0,0 +1,299 @@
...
+	setRows: 4,
+	setColumns: 4,
+	setLimit: 6,
+	setMultiplier: 4,
+	numberOfCombinations: 1600,
+	setSize: 500,
+}
+
+function swap(arr,i1,j1,i2,j2) {
+	t = arr[i1][j1];
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:157]: OK, I understood

File: js/game.js [hunk 25]
```diff
@@ -0,0 +1,299 @@
...
+	arr[i1][j1] = arr[i2][j2];
+	arr[i2][j2] = t;
+}
+
+window.onload = function() {
+	PopUpHide();
+	box = document.getElementById("box");
+	createNumbers();
+	document.getElementById("reset").onclick = () => {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:166]: OK, I understood

File: js/game.js [hunk 26]
```diff
@@ -0,0 +1,299 @@
...
+		createNumbers();
+		ClearСlock();
+		startTimer();
+	}
+}
+console.log(imageSrc)
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:172]: OK, I understood

File: js/game.js [hunk 27]
```diff
@@ -0,0 +1,299 @@
...
+
+
+
+function cellClick(event) {
+	var event = event || window.event,
+		element = event.srcElement || event.target,
+		emptyElement = document.getElementById(emptyRow + " " + emptyColumn),
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:179]: OK, I understood

File: js/game.js [hunk 28]
```diff
@@ -0,0 +1,299 @@
...
+		//get the number of the row and column at the intersection of which the cell is located.
+		i = element.id.charAt(0),
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:181]: OK, I understood

File: js/timer.js [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+var h = 1,
+  m = 1,
+  tm = 1,
+  s = 0,
+  ts = 0,
+  ms = 0,
+  init = 0;
+
+//функция для очистки поля
+function ClearСlock() {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/timer.js:15]: OK, I understood

File: js/game.js [hunk 30]
```diff
@@ -0,0 +1,299 @@
...
+			if (cell.innerHTML == '') {
+				cell.style.backgroundColor = "#56526a"
+			}
+			//added row in table
+			row.appendChild(cell);
+	}
+	board.appendChild(row);
+}
+//check if the element has a table
+	if(box.childNodes.length == 1)
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:296]: OK, I understood

File: js/timer.js [hunk 31]
```diff
@@ -0,0 +1,103 @@
...
+  if (dh > 0) {
+    if (dh < 10) {
+      dh = '0' + dh;
+    }
+  } else {
+    dh = '00';
+  }
+  readout = dh + ':' + dm + ':' + ds;
+  document.myForm.stopwatch.value = readout;
+  clocktimer = setTimeout("StartTIME()", 1);
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/timer.js:89]: OK, I understood

File: js/game.js [hunk 32]
```diff
@@ -0,0 +1,299 @@
...
+			// left
+			case 3: if(emptyColumn != 0) swap(arr,emptyRow,emptyColumn,emptyRow,--emptyColumn);
+		}
+
+//create a board
+var board = document.createElement("board");
+board.style.display = "flex";
+board.style.flexDirection = "column";
+board.style.justifyContent = "center";
+board.style.alignItems = "center";
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:258]: OK, I understood

File: js/game.js [hunk 33]
```diff
@@ -0,0 +1,299 @@
...
+			cell.style.display = "flex";
+			cell.style.justifyContent = "center";
+			cell.style.alignItems = "center";
+			cell.style.color = "white";
+			cell.style.border = "solid white 1px"
+			cell.style.fontSize = "40px";
+			cell.style.minWidth = "55px";
+			cell.style.minHeight = "55px";
+			cell.style.margin = "2px";
+			cell.style.backgroundColor = "#4d495f"
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:279]: OK, I understood

File: js/game.js [hunk 34]
```diff
@@ -0,0 +1,299 @@
...
+var nav_right = document.createElement("nav");
+var h2_right = document.createElement("h2");
+var list_right = document.createElement("ul");
+var scoreButton = document.createElement("li");
+var top_score = document.createElement("li");
+//create central box
+var centralBox = document.createElement("div");
+var infoBox = document.createElement("div");
+var box = document.createElement("div");
+var form = document.createElement("form");
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/game.js:30]: OK, I understood

File: js/timer.js [hunk 35]
```diff
@@ -0,0 +1,103 @@
...
+  s = 0;
+  ts = 0;
+  ms = 0;
+  init = 0;
+  readout = '00:00:00';
+  document.myForm.stopwatch.value = readout;
+}
+
+//функция для старта секундомера
+function StartTIME() {
```
[the remark this answers is not in the document]
Comment (andrei-roh) [reply js/timer.js:29]: OK, I understood

File: js/boards/bigBoard.js (hunk 1 above)
Comment (andrei-roh) [reply js/boards/bigBoard.js:4]: OK, I understood

**Pull request merged**